src/Microsoft.ApiDesignGuidelines.Analyzers/UnitTests/DoNotDecreaseInheritedMemberVisibilityTests.cs

```diff
@@ -12,12 +10,295 @@ public class DoNotDecreaseInheritedMemberVisibilityTests : DiagnosticAnalyzerTes
     {
         protected override DiagnosticAnalyzer GetBasicDiagnosticAnalyzer()
         {
-            return new BasicDoNotDecreaseInheritedMemberVisibilityAnalyzer();
+            return new DoNotDecreaseInheritedMemberVisibilityAnalyzer();
         }
 
         protected override DiagnosticAnalyzer GetCSharpDiagnosticAnalyzer()
         {
-            return new CSharpDoNotDecreaseInheritedMemberVisibilityAnalyzer();
+            return new DoNotDecreaseInheritedMemberVisibilityAnalyzer();
+        }
+
+        [Fact]
+        public void DecreaseCSharpMemberVisibility()
+        {
+            VerifyCSharp(
+@"public class BaseClass
+{
+    public int MyProp3;
+}
+
+public class DerivedClass : BaseClass
+{
+    private new int MyProp3;
+}", GetCSharpCA2222RuleNameResultAt(8, 21));
+
+            VerifyCSharp(
+@"public class BaseClass
+{
+    public int MyProp3 { get; set; }
+}
+
+public class DerivedClass : BaseClass
+{
+    internal new int MyProp3 { get; set; }
+}", GetCSharpCA2222RuleNameResultAt(8, 22), GetCSharpCA2222RuleNameResultAt(8, 32), GetCSharpCA2222RuleNameResultAt(8, 37));
```

Comment:
This is a bit tricky because FxCop actually only flagged methods, so it wouldn't report on the property in this case (just on the accessors). Mimicking that behavior exactly seems wrong, though, since this can't be fixed at the accessor level. 

I like your idea of only flagging accessors with explicitly specified accessibilities, but that moves us a bit farther away from FxCop compat.

What's the guidance for tweaking behavior to make things more code-fix-friendly? Is that ok, or do we want to be as close to FxCop as possible? (If we need to be _exactly_ like FxCop, I should probably not fire on properties and just not provide code fixes for these cases).